dex/tests/test.sh

```diff
@@ -1,39 +1,42 @@
 #!/bin/sh
+#/ Usage: test.sh <pkg_ident>
+#/
+#/ Example: test.sh core/dex/2.17.0/20190531065119
+#/
 
-TESTDIR="$(dirname "${0}")"
-PLANDIR="$(dirname "${TESTDIR}")"
-SKIPBUILD=${SKIPBUILD:-0}
+set -euo pipefail
+
+if [[ -z "${1:-}" ]]; then
+  grep '^#/' < "${0}" | cut -c4-
+  exit 1
+fi
 
+TESTDIR="$(dirname "${0}")"
 source "${TESTDIR}/helpers.bash"
 
-hab pkg install core/bats --binlink
+export TEST_PKG_IDENT="${1}"
+hab pkg install "${TEST_PKG_IDENT}"
 
 hab pkg install core/busybox-static
 hab pkg binlink core/busybox-static nc
-hab pkg install -b core/jq-static core/curl
+hab pkg install core/bats core/curl core/jq-static --binlink
 
-# Wait for supervisor to start
-echo "Waiting for supervisor to start"
-wait_listen tcp 9632 30
+hab sup term
```

Comment:
I don't think we want to be stopping and starting the supervisor in the test.  In the context of CI we should always have a clean system. If we're worried about having a "clean" (no other services running) supervisor, starting and stopping isn't sufficient, we'd also need to unload any other services and that feels out of scope for what these tests should be doing.  

I also suspect that starting it in the background without the setsid & disown that `sup-run` does may cause the CI to hang until timeout as there will still be jobs associated with the process.

Reply:
happy to change it. Just remove the `hab sup term` or something else? I'm happy to follow any examples, I had been taking this from the last iteration of service-tests I had worked with, `opa/test/test.sh`

Reply:
Let me check and make sure I didn't miss a decision on this.  That feels like an anti-pattern to me, but I want to make sure we're consistent with our testing patterns.

Reply:
While I'm tracking this down,  if this goes green in CI I'm :+1: on merging.  We can do a follow-on PR to align on what I learn, if needed.

Reply:
Finally green on buildkite -- had to change another thing. Also re-added the supervisor startup bits, as it wouldn't have been accessible otherwise when the tests were run.

Reply:
@smacfarlane thanks for bearing with me 😃

Reply:
@smacfarlane I believe @predominant would disagree with this and would say that we need to do `hab sup term / hab sup run &`. What is the right course of action here?

Reply:
While I understand where you might get this assumption, @echohack - Its actually not true.

I've included this same `hab sup term` and restart sequence in my pull requests. But I feel a bit dirty each time. Its not a good pattern, but I've had low motivation to fix it while I have been working on other things.

I think the solution is somewhere between assuming a clean environment and killing the supervisor.

* Is a supervisor running?
 * Yes: Good. Leave it as is.
 * No: Start one (properly, not with just `&`)

In CI this produces the best situation - No supervisor on startup in `hab studio run ... ` causes a supervisor to start, and then tests are run.

In an interactive studio, the check shows a supervisor already running and so does nothing. In this case, it is up to the developer/user to ensure no interfering services are running, and we still recommend that `hab studio run ... ` is done prior to a pull request to ensure it works in a fresh environment.

Reply:
😃 How about this? https://github.com/habitat-sh/core-plans/pull/2772/files#diff-85b160b35bb84807f2d72298be2450e8R24

Reply:
@srenatus I think that's acceptable for now. Based on Graham's feedback here as well, I'll go ahead and approve this. :)